crates/sui-json-rpc/src/unit_tests/rpc_server_tests.rs

```diff
@@ -1189,11 +1188,11 @@ async fn test_delegation_with_locked_sui() -> Result<(), anyhow::Error> {
     // Check StakedSui object
     let staked_sui: Vec<DelegatedStake> = http_client.get_delegated_stakes(*address).await?;
     assert_eq!(1, staked_sui.len());
-    assert_eq!(1000000, staked_sui[0].staked_sui.principal());
-    assert_eq!(Some(20), staked_sui[0].staked_sui.sui_token_lock());
+    assert_eq!(1000000, staked_sui[0].delegations[0].principal);
+    assert_eq!(Some(20), staked_sui[0].delegations[0].token_lock);
 
     assert!(matches!(
-        staked_sui[0].delegation_status,
+        staked_sui[0].delegations[0].status,
```

Comment:
Since there are quite some changes to the logic of `get_delegated_stakes`, we should probably add some tests for when a delegator delegates with multiple validators and check that all the returned delegated stakes are correct?